runtime/lua/vim/ui.lua

```diff
@@ -104,4 +104,45 @@ function M.input(opts, on_confirm)
   end
 end
 
+--- Opens a URI by sending it to the system's default handler.
+--- This function utilizes `xdg-open`, `wslview`, or `open` commands depending on
+--- the operating system to open the provided URI.
+---
+---
+---@param uri string Uri to be opened
+---   TODO: uri should accept vim.uri.Uri
```

Comment:
```suggestion
```

There is currently no `vim.uri.Uri` class. I think this can be removed for now.

Reply:
@justinmk is accepting a Uri class really that important? There are no actions performed on it and all the handlers deal with parsing the "uri".